assets/js/admin/marketplace-suggestions.js

```diff
@@ -0,0 +1,360 @@
+/* global marketplace_suggestions ajaxurl */
+( function( $, marketplace_suggestions ) {
+	$( function() {
+		if ( 'undefined' === typeof marketplace_suggestions ) {
+			return;
+		}
+
+		// Stand-in wcSettings.recordEvent for until tracks PR is merged.
+		// https://github.com/woocommerce/woocommerce/pull/22746
+		window.wcSettings = window.wcSettings || {};
+		window.wcSettings.recordEvent = window.wcSettings.recordEvent  || function() { };
+
+		// Tracks events sent in this file:
+		// - marketplace_suggestion_displayed
+		// - marketplace_suggestion_clicked
+		// - marketplace_suggestion_dismissed
+		// All are prefixed by {WC_Tracks::PREFIX}.
+		// All have one property for `suggestionSlug`, to identify the specific suggestion message.
+
+		// Dismiss the specified suggestion from the UI, and save the dismissal in settings.
+		function dismissSuggestion( suggestionSlug ) {
+			// hide the suggestion in the UI
+			var selector = '[data-suggestion-slug=' + suggestionSlug + ']';
+			$( selector ).fadeOut( function() {
+				$( this ).remove();
+			});
+
+			// save dismissal in user settings
+			jQuery.post(
+				ajaxurl,
+				{
+					'action': 'woocommerce_add_dismissed_marketplace_suggestion',
+					'_wpnonce': marketplace_suggestions.dismiss_suggestion_nonce,
+					'slug': suggestionSlug,
+				}
+			);
+
+			window.wcSettings.recordEvent( 'marketplace_suggestion_dismissed', {
+				suggestionSlug: suggestionSlug
+			} );
+		}
+
+		// Render DOM element for suggestion dismiss button.
+		function renderDismissButton( suggestionSlug ) {
```

Comment:
Not related to this specifically but I noticed the suggestions were dissmissble on blank slate and you can dismiss all instead nothing is left.

Perhaps those ones don't need to be dismissable - they won't be show long anyway, once you've added a product they are forever hidden?

Reply:
That's a reasonable suggestion – dismissability is controlled from the JSON data (per suggestion), so we can iterate on that outside of this PR/woo code base.